sdks/python/apache_beam/dataflow_test.py

```diff
@@ -109,11 +109,12 @@ def test_par_do_with_side_input_as_keyword_arg(self):
 
   @attr('ValidatesRunner')
   def test_par_do_with_do_fn_object(self):
-    class SomeDoFn(DoFn):
+    class SomeDoFn(NewDoFn):
       """A custom DoFn for a FlatMap transform."""
 
-      def process(self, context, prefix, suffix):
-        return ['%s-%s-%s' % (prefix, context.element, suffix)]
+      def process(self, prefix, element=NewDoFn.ElementParam,
```

Comment:
Makes sense. Two follow up questions then:

1) Should we change functions like https://github.com/apache/beam/blob/python-sdk/sdks/python/apache_beam/io/iobase.py#L752 to have the second argument tagged as `SideInputParam` ? 

2) Do we want to differentiate between deferred and non-deferred side inputs or it should not matter. Eg. https://github.com/apache/beam/blob/python-sdk/sdks/python/apache_beam/io/iobase.py#L776